components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java

```diff
@@ -167,33 +169,124 @@ public Response apisApiIdGraphqlSchemaGet(String apiId, String ifNoneMatch, Stri
     }
 
     @Override
-    public Response apisApiIdCommentsCommentIdDelete(String commentId, String apiId, String ifMatch, MessageContext messageContext) {
-        // do some magic!
-        return Response.ok().entity("magic!").build();
+    public Response addCommentToAPI(String apiId, CommentDTO body, MessageContext messageContext) {
+        String username = RestApiUtil.getLoggedInUsername();
+        String requestedTenantDomain = RestApiUtil.getLoggedInUserTenantDomain();
+        try {
+            APIConsumer apiConsumer = RestApiUtil.getLoggedInUserConsumer();
+            APIIdentifier apiIdentifier = APIMappingUtil.getAPIIdentifierFromUUID(apiId, requestedTenantDomain);
+
+            Comment comment = CommentMappingUtil.fromDTOToComment(body, username, apiId);
+            int createdCommentId = apiConsumer.addComment(apiIdentifier, comment, username);
+            Comment createdComment = apiConsumer.getComment(apiIdentifier, createdCommentId);
+            CommentDTO commentDTO = CommentMappingUtil.fromCommentToDTO(createdComment);
+
+            String uriString = RestApiConstants.RESOURCE_PATH_APIS + "/" + apiId +
+                    RestApiConstants.RESOURCE_PATH_COMMENTS + "/" + createdCommentId;
+            URI comments = new URI(uriString);
```

Comment:
Shall we rename URI `comments` to `uri`?

Reply:
fixed